Project/Source/Components/UI/ComponentEventSystem.cpp

```diff
@@ -14,10 +16,26 @@
 #define JSON_TAG_FIRST_SELECTED_ID "FirstSelectedId"
 
 ComponentEventSystem ::~ComponentEventSystem() {
+	ComponentEventSystem* eventSystem = App->userInterface->GetCurrentEventSystem();
+	if (eventSystem != nullptr) {
+		if (eventSystem->GetID() == GetID()) {
+			App->userInterface->SetCurrentEventSystem(0);
+		}
+	}
+	else {
+		App->userInterface->SetCurrentEventSystem(0);
+	}
+	eventSystem = App->userInterface->GetCurrentEventSystem();
+	LOG("deleted %u", GetID());
```

Comment:
I don't undestand what happens here..
Can you explain me a bit? :D

Reply:
This is probably no longer necesary since we capped the ComponentEventSystem to one single instance, this code managed the fact that upon destroying a ComponentEventSystem, the Module would look for a substitute of an EventSystem among the existing ones, if any. Will be removed before comfirming the PR